pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -83,10 +90,54 @@ func (s *preFilterState) Clone() framework.StateData {
 // failed by this plugin schedulable.
 func (pl *NodeAffinity) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
 
+// isSchedulableAfterNodeChange is invoked whenever a node changed. It checks whether
+// that change made a previously unschedulable pod schedulable. It errs on the side of
+// letting a pod scheduling attempt happen.
+func (pl *NodeAffinity) isSchedulableAfterNodeChange(pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	if newObj == nil {
+		// Deletes don't make a pod schedulable.
+		return framework.QueueSkip
+	}
+
+	modifiedNode, ok := newObj.(*v1.Node)
+	if !ok {
+		// Shouldn't happen.
+		pl.logger.Error(nil, "unexpected new object in isSchedulableAfterNodeChange", "obj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	if oldObj == nil {
+		pl.logger.V(4).Info("node was created", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+		return framework.QueueAfterBackoff
+	}
+
+	// Modifications may or may not be relevant. If the entire
+	// labels is as before, then something else must have changed
+	// and we don't care.
+	originalNode, ok := oldObj.(*v1.Node)
+	if !ok {
+		// Shouldn't happen.
+		pl.logger.Error(nil, "unexpected old object in isSchedulableAfterNodeChange", "obj", oldObj)
+		return framework.QueueAfterBackoff
+	}
+	if apiequality.Semantic.DeepEqual(originalNode.Labels, originalNode.Labels) {
```

Comment:
We can make isSchedulableAfterNodeChange more specific. 
If the label, that isn't related to the Pod's NodeAffinity, is updated, we can also return skip.

Reply:
Is it duplicated with the filter logic if check labels and  pod's NodeAffinity? https://github.com/carlory/kubernetes/blob/fa80451432acc2cd2adb8d683c2424c5a6783bae/pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go#L198-L208

Reply:
Yes, and that's fine. That is definitely a benefit of QueueHint, we can ignore the event based on more specific checking and reduce the possibility the Pod is rejected in the scheduling cycle. The logic of QueueHint may be duplicated with what the plugin does in the scheduling cycle, but it should be accepted (as long as the logic isn't super heavy.)